### DISCUSSION TITLE
Updated the data for p0 for PV, but no impact on TDS results

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#258] [General]
Hello @cuihantao, @jinningwang,

I want to see the impact of **p0** of **PV** on TDS results in DGPRCT1 (Togglers are disabled) use case. I followed the following steps:
1. Ran the TDS for 5 seconds.
2. Updated the **p0 value of PV** using the below command:
     `ss.PV.alter("p0", 2, 0.1)`
3. Continued the TDS for 10 seconds.
4. Reset the **p0 value of PV** to original using the alter command.
5. Continued the TDS for 20 seconds.
6. On plotting the graph for BusFreq, there were all straight lines as shown below:

![Untitled picture](https://user-images.githubusercontent.com/96423656/151596159-59d1ba78-cb2b-4ac5-adef-b6ac59b15c08.png)

FYI, I have also tried updating **p** value of **PV** on step 2, but no impact on the plot.
Please guide me, where I'm going wrong.

Thanks

**Comment by cuihantao** [comment]
@jinningwang might be able to help you.

**Comment by jinningwang** [comment]
Hi there,

I understand you want to see the impact of p0 of PV on TDS results.
Parameter p0 refers to the active power setpoint, which is used as the initial value of power flow calculation. **TDS** will take the power flow solution and do the dynamic simulation. **Once the dynamic simulation starts, p0 will no longer impact the system.** 
Changing the p0 value before running TDS should be able to make an impact.

Regards,
Jinning

**Reply by TheASingh** [reply]
Hi @jinningwang,

Thanks for your response. I have the following query:

Is there any way to change the p0(active power) value during simulation?

Below is the scenario in which I'm interested:

1. Run the TDS for 5 seconds.
2. Update the p0 value of PV.
3. Continue the TDS for 10 seconds.
4. Reset the p0 value of PV to original value
5. Continue the TDS for 20 seconds.
6. Plot the graph for BusFreq to check the impact of p0

Your response will be highly appreciable.

Thanks,
A Singh

**Reply by jinningwang** [reply]
Definitely, you can change the p0 value during the simulation, it is just not impacting the system at all. The device ``PV`` (from group ``StaticGen``) is not used anymore once the dynamic simulation starts.
The device ``PV`` is used to do the power flow calculation. Then the calculated initial power flow will be used by ``SynGen`` and some other renewable power generation like ``RenGen`` or ``DG``.
In the dynamic simulation, you can change the active power setpoint of a ``GENROU``, by altering the variable ``Paux`` of its connected governor. For more detail, you can refer to ``TGOV1`` documentation.

Regards,
Jinning

**Reply by TheASingh** [reply]
@jinningwang,

Thanks for your response, I tried it but the plots are still showing the straight lines. 

I made the following alterations during simulation:
i. Altered `Paux` variable of all `TGOV1` items

(Even I tried the following as well)
ii. Altered `p0` of all `GENROU `items
iii. Altered `pe` of all `GENROU `items

Please guide me how can I see variations in frequency plots. It will be really appreciable.

Thanks,
A Singh

**Reply by jinningwang** [reply]
Sorry for the previous misleading, I checked the documentation and the accurate variable name should be `paux` as you can find in: https://docs.andes.app/en/stable/groupdoc/TurbineGov.html?highlight=TGOV1#tgov1

Another thing is that I'm not sure in the case you used the governor model is `TGOV1`, which is just an example for my explanation. Make sure you altered the correct model in your case. 

After you alter the value of `paux`, you can 1) check the corresponding generator output (two variables `Pe` and `omega` can be helpful). 2) if not, check whether the output of the governor (`pout`) is changed. 3) if not, check whether the variable is altered successfully.

One more thing, you can search the model diagram with 'PowreWorld + [model name]', which will be helpful when you read the model documentation. 

Regards,
Jinning

**Reply by TheASingh** [reply]
Thanks @jinningwang for the reply,

I have used **ieee14_dgprct1** use case. 

**First try**:
As per the original input file which has two Togglers for PQ_12:
After updated `paux` for `TGOV1`, there was change in `Pe`, `omega`, `pout` values.

**Second try**:
I updated the input file and set the `u` to zero for both the Togglers for PQ_12:
After updated `paux` variable for `TGOV1`, there was **no** change in `Pe`, `omega`, `pout` values.





**First Query**:
Please guide me, If I'm doing in correct way in the second try.

**Second query**:
If I don't have any `TGOV1 `in another use case, in that case how can I change the active power setpoint of a `GENROU` in dynamic simulation?

Regards,
A Singh

**Reply by jinningwang** [reply]
Hi Singh, @TheASingh 

I checked the documentation and tested the case you said, I realized that I have previously misled you a little bit.

Q1: In the given exaple you can find that, the setpoint of governor is modified by `paux0`: https://github.com/cuihantao/andes/blob/develop/examples/8.%20change-setpoints.ipynb
Sorry for my previous misled you playing with `paux`.

The detailed reason is that: the `alter` method defined in the model is used to modify the values of `service` or `parameter`, as described in the documentation: https://docs.andes.app/en/stable/moduledoc/andes.core.html?highlight=alter#andes.core.model.Model.alter
It means that we should modify `paux0` rather than `paux`, since `paux0` is a `ConstService` whereas `paux` is a `Algeb`, as described in the documentaion: https://docs.andes.app/en/stable/groupdoc/TurbineGov.html#tgov1

Q2: `paux0` is the common `ConstService` used by group `TurbineGov` as the initial value of `paux`. It means that as long as there is a governor model, you can change the active power setpoint by altering the value of `paux0`. It is common and reasonable that a generator is equiped with a governor.

Hope this helps you out.

Regards,
Jinning

**Reply by TheASingh** [reply]
Thank you very much @jinningwang,

Now, after altering `paux0`, I'm able to see the impact on the plots. 
Please clear my doubts on the following items:

1. Before altering the `paux0` value, I checked all the `TGOV1s` were having zero values for `paux0 `as shown below.
   
   ![image](https://user-images.githubusercontent.com/96423656/152027639-1a2e4cd6-366d-417e-b75f-42bf44c163ea.png)

    And `p0` of the PV is let's say is `0.3`. Then in order to **decrease** the active power setpoint, what value should I set to `paux0` (as it already is 
    zero)?

2. If there is no governor attached to a generator, is it possible to change its active power setpoint dynamically?

3. Also in the DGPRCT1 use case, there are only three governors but generators are five. It means two generators do not have governors. Then how their active power setpoint can be updated?

Regards,
A Singh

**Reply by jinningwang** [reply]
1. It is also acceptable to set negative values for `paux0`. You can check the control diagram of TGOV1, which will give you a better clue of the meaning of those variables: https://www.powerworld.com/WebHelp/Content/TransientModels_PDF/Generator/Governor/Governor%20TGOV1.pdf
2. It is possible, by playing with the `tm0` which is the initial value of mechanical torque: https://docs.andes.app/en/stable/groupdoc/SynGen.html?highlight=GENROU
3. Same to Q2.

Jinning

**Reply by TheASingh** [reply]
Thanks Jinning, it worked for me.

I changed the value of `tm0` of `GENROU_2` from **0.4** to **0.9**. When I plot a graph for `Pe`, lines were oscillating as shown below.

Is it correct? If yes, then please explain little bit for my understanding.


![image](https://user-images.githubusercontent.com/96423656/152186827-14bcd2bf-a0ba-40c6-80f1-bb97e84d450d.png)


Regards,
A Singh

**Reply by jinningwang** [reply]
I think it is correct. The oscillation is reasonable because the power increase is large.
I don't know what happened at 10s, it looks like protection is triggered or something is tripped.

Regards,
Jinning
